processing/src/main/java/org/apache/carbondata/processing/newflow/CarbonDataLoadConfiguration.java

```diff
@@ -0,0 +1,185 @@
+package org.apache.carbondata.processing.newflow;
+
+import java.util.Iterator;
+
+import org.apache.carbondata.core.carbon.AbsoluteTableIdentifier;
+
+public class CarbonDataLoadConfiguration {
```

Comment:
It seems this configuration is quite complex, I think it is because it contains configuration for all steps. 
Can we just have a simple `Map` as the configuration and let the `Step` decide what to keep in it?